dataset/src/test/scala/frameless/ColTests.scala

```diff
@@ -54,4 +56,16 @@ class ColTests extends TypedDatasetSuite {
     check(prop[X2[Int, Int]] _)
     check(prop[X2[X2[Int, Int], Int]] _)
   }
+
+  test("schema wrapper test: wrapper activated") {
+    val t = TypedDataset.create(1 to 2)
+    t.col[Int]('_1)
```

Comment:
I'm not totally convinced about that, it feels a bit ad hoc. I feel that wrapping types of non case class Dataset's in a Tuple1 is a bit of a hack, it would be cleaner to either:

1. Always require case classes for Dataset (and wrap stuff in Tuple1 accordingly). This might also be what we want in the long term if we switch to a record representation instead to keep labels around. At the same time, this would be big divergence from vanilla...

2. Add some sort of "top level" selector that works symmetrically on case classes and ground types

Reply:
For 1, do you mean that we require the schema to be `TypedDatset[(A)]` instead of `TypedDataset[A]`. This would be a divergence from vanilla. Also performance wise, getting back an`Array[Int]` (after a .collect()) is not the same as getting back an `Array[Tuple1[Int]]`.

For 2, I am not sure I understand that so let me try to re-phrase and correct me if I am wrong: Have an operator that explicitly converts `TypedDataset[A]` to `TypedDatset[Tuple1[A]]` so that `select('_1)` etc. would work. That would make it more explicit that you want to work with `TypedDatset[Tuple1[A]]` rather than implicitly converting the type?

Reply:
1. :+1: 
2. No, I was more thinking something like a `***` selector to be able to do `select(***)` instead of `select('_1)` on a `TypedDataset[Int]`, but would also work to talk to select any `A` from `TypedDatset[A]` (unlike what you are proposing, IIUC)